ci/setup-build-test-gdk.ps1

```diff
@@ -3,22 +3,61 @@ param(
   [string] $msbuild_exe = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe",
   [string] $build_home = (Get-Item "$($PSScriptRoot)").parent.parent.FullName, ## The root of the entire build. Should ultimately resolve to "C:\b\<number>\".
-  [string] $unreal_path = "$build_home\UnrealEngine",
-  [string] $test_repo_branch = "master",
-  [string] $test_repo_url = "https://github.com/spatialos/UnrealGDKTestGyms.git",
-  [string] $test_repo_relative_uproject_path = "Game\GDKTestGyms.uproject",
-  [string] $test_repo_map = "EmptyGym"
+  [string] $unreal_path = "$build_home\UnrealEngine"
 )
 
+class TestSuite {
+  [ValidateNotNullOrEmpty()][string]$test_repo_url
+  [ValidateNotNullOrEmpty()][string]$test_repo_branch
+  [ValidateNotNullOrEmpty()][string]$test_repo_relative_uproject_path
+  [ValidateNotNullOrEmpty()][string]$test_repo_map
+  [ValidateNotNullOrEmpty()][string]$test_project_root
+  [ValidateNotNullOrEmpty()][string]$test_results_dir
+  [ValidateNotNullOrEmpty()][string]$tests_path
+  [bool]$run_with_spatial
+
+  TestSuite([string] $test_repo_url, [string] $test_repo_branch, [string] $test_repo_relative_uproject_path, [string] $test_repo_map, [string] $test_project_root, [string] $test_results_dir, [string] $tests_path, [bool] $run_with_spatial) {
+    $this.test_repo_url = $test_repo_url
+    $this.test_repo_branch = $test_repo_branch
+    $this.test_repo_relative_uproject_path = $test_repo_relative_uproject_path
+    $this.test_repo_map = $test_repo_map
+    $this.test_project_root = $test_project_root
+    $this.test_results_dir = $test_results_dir
+    $this.tests_path = $tests_path
+    $this.run_with_spatial = $run_with_spatial
+  }
+}
+
+[string]$test_repo_branch = "master"
+[bool]$slow_networking_tests = $False
+
 # Allow overriding testing branch via environment variable
 if (Test-Path env:TEST_REPO_BRANCH) {
   $test_repo_branch = $env:TEST_REPO_BRANCH
 }
+# Allow overriding running slow networking tests
+if ( ((Test-Path env:SLOW_NETWORKING_TESTS) -And ($env:SLOW_NETWORKING_TESTS -eq "true")) -Or ($env:NIGHTLY_BUILD -eq "true")) {
+  $slow_networking_tests = $True
+}
+
+$tests = @(
+  [TestSuite]::new("git@github.com:improbable/UnrealGDKEngineNetTest.git", "$test_repo_branch", "Game\EngineNetTest.uproject", "NetworkingMap", "NetworkTestProject", "FastTestResults", "SpatialGDK+/Game/SpatialNetworkingMap", $True)
+)
+
+if($slow_networking_tests) {
+  $tests[0].tests_path += "+/Game/NetworkingMap"
+  $tests[0].test_results_dir = "SpatialTestResults"
+  $tests += [TestSuite]::new("git@github.com:improbable/UnrealGDKEngineNetTest.git", "$test_repo_branch", "Game\EngineNetTest.uproject", "NetworkingMap", "NetworkTestProject", "VanillaTestResults", "/Game/NetworkingMap", $False)
+}
 
 . "$PSScriptRoot\common.ps1"
 
 # Guard against other runs not cleaning up after themselves
-& $PSScriptRoot"\cleanup.ps1"
+foreach ($test in $tests) {
```

Comment:
In other classes this is `Foreach`............. I dunno man we need a linter

Reply:
Applied some formatting to all powershell files